src/delivery_problems/mod.rs

```diff
@@ -0,0 +1,289 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Bounce and complaint handling.
+
+use std::{collections::HashMap, time::SystemTime};
+
+use serde::{
+    de::{Deserialize, Deserializer, Error as DeserializeError, Unexpected},
+    ser::{Serialize, Serializer},
+};
+
+use app_errors::{AppErrorKind, AppResult};
+use auth_db::Db as AuthDb;
+use email_address::EmailAddress;
+use queues::notification::{BounceSubtype, BounceType, ComplaintFeedbackType};
+use settings::{BounceLimit, BounceLimits, Settings};
+
+#[cfg(test)]
+mod test;
+
+/// Bounce/complaint registry.
+///
+/// Currently this writes to
+/// both Redis and
```

Comment:
It doesn't appear to write to Redis, is this comment out-of-date?

Reply:
Oh whoops, it's an extraction from the branch that *does* write to Redis, which is my next PR. So, yes, out of date but not in the conventional direction. I'll remove it from this diff.